site/blog/unlisted/subqueries.md

```diff
@@ -68,4 +75,12 @@ WHERE cost_center_id = (
 );
 ```
 
-In conclusion, there are several advantages to using subqueries instead of joins in PostgreSQL. Whether you are dealing with large datasets, need to improve the readability of your code, or want to increase your flexibility and simplify maintenance, subqueries may be the better option for your project. Consider using subqueries the next time you need to combine data from multiple tables in PostgreSQL, and see the benefits for yourself!
\ No newline at end of file
+### What about Performance?
+
+The last, but not least important, factor to consider is performance. There is no clear winner in terms of performance. Performance varies a lot depending on the data model, data volume, and query design. Joins can be resource-intensive, especially when dealing with large data sets or complex queries. Subqueries, on the other hand, can be faster and more efficient in some cases. It is essential to evaluate the complexity of the query and the size of the data sets involved to determine which technique is more appropriate.
+
+Subqueries can sometimes be optimized and executed more efficiently than joins, especially when dealing with large amounts of data. In PostgreSQL, subqueries can be executed in parallel, which can significantly improve performance while a single query might not be able to utilize all the cores available. By breaking down the query into smaller parts, the database can use more computing resources, if available, but also any indexing and caching mechanisms to optimize the performance of the query. The main consideration for performance gains in subqueries is that the output of the sub-query itself will not be indexed, so the cardinality of the output needs to be much smaller than the inputs for the performance benefit to occur. In contrast, joins *can* be slower and more resource-intensive, especially when dealing with complex queries or large datasets. However, joins can mitigate the processing burden overhead on the database by replacing multiple queries with one join query. This in turn makes better use of the database's ability to search through, filter, and sort records. Having said that, as you add more joins to a query, the database server has to do more work, which translates to slower data retrieval times. Much like any performance computing optimization or [Amdal's Law](https://en.wikipedia.org/wiki/Amdahl%27s_law), more parallelization is not always the correct answer.
+
+PostgreSQL offers a query analyzer that can help you better understand the performance characteristics of a given query. Simply prepend `EXPLAIN ANALYZE` before any given query. Historically, explicit joins have better performance, but query optimizers are getting better over time so it is best to write queries first in a logically coherent way, and then restructure if performance is a problem.
```

Comment:
This blog post would benefit from looking at two queries and picking at their `EXPLAIN` outputs, I think.

Reply:
The thing is that the `EXPLAIN` output from a query is highly dependent on the data in the database

Reply:
That's true, but at least having an example of what that output can look like and how to interpret what it's telling you can be useful, even if the details differ for the users? We can just tell them that.